src/universal/components/BaseButton.js

```diff
@@ -4,33 +4,41 @@ import styled from 'react-emotion'
 import ui from 'universal/styles/ui'
 import PlainButton from 'universal/components/PlainButton/PlainButton'
 import withInnerRef from 'universal/decorators/withInnerRef'
+import elevation from 'universal/styles/elevation'
 
-const ButtonRoot = styled(PlainButton)(({size, depth, disabled, pressedDown}) => {
-  const hasDepth = depth || depth === 0
-  const boxShadow = hasDepth ? ui.shadow[depth] : 'none'
-  const hoverDepth = hasDepth ? ui.shadow[depth + 2] : ui.shadow[1]
-  const pressedDepth = hasDepth ? ui.shadow[depth + 1] : ui.shadow[0]
-  const stateDepth = pressedDown ? pressedDepth : hoverDepth
-  return {
-    // size is easy to override, it adds: fontSize, lineHeight, padding
-    ...ui.buttonSizeStyles[size],
-    display: 'block',
-    border: '.0625rem solid transparent',
-    boxShadow,
-    textAlign: 'center',
-    transform: pressedDown && 'translate(0, .125rem)',
-    transition: `
-        box-shadow ${ui.transition[0]},
-        transform ${ui.transition[0]}
-      `,
-    userSelect: 'none',
-    whiteSpace: 'nowrap',
-    ':hover,:focus,:active': {
-      boxShadow: disabled ? 'none' : stateDepth,
-      outline: pressedDown && 0
+const ButtonRoot = styled(PlainButton)(
+  ({disabled, elevationResting, elevationHovered, pressedDown, size}) => {
+    const hasDepth = elevationResting && elevationHovered
+    return {
+      // size is easy to override, it adds: fontSize, lineHeight, padding
+      ...ui.buttonSizeStyles[size],
+      alignItems: 'center',
+      border: '.0625rem solid transparent',
+      boxShadow: !disabled && hasDepth ? elevation[elevationResting] : undefined,
```

Comment:
why is `hasDepth` necessary?
if the resting elevation is 0 , this will be undefined.
if there is no hover elevation, it is also undefined, even though it has a resting elevation. that seems weird